### PR TITLE
chore: add a better discord flag from env

### PR DESCRIPTION
**Pull request by Feichtmeier**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -96,7 +97,7 @@ parts:
       # work around pub get stack overflow # https://github.com/dart-lang/sdk/issues/51068#issuecomment-1396588253
       set +e
       dart pub get
-      flutter build linux --release -v
+      flutter build linux --release -v --dart-define=ALLOW_DISCORD_RPC=${ALLOW_DISCORD_RPC}
```
[the remark this answers is not in the document]
Comment (Feichtmeier) [reply snap/snapcraft.yaml:100]: ok and what about line 22 ?
how will you then toggle it on and off? you just do this locally now?

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:100]: So, the rationale is that, it'll check if the ALLOW_DISCORD_RPC says true or false in the environment.. if it's not defined then it's false, else true.. That's it, nothing extra.

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:100]: Okay, let me try a local snap once with this patch

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: We have a problem... didn't work.. let me check the source code again.

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: runtime environment variables doesn't seem to work in dart

[the remark this answers is not in the document]
Comment (Feichtmeier) [reply snap/snapcraft.yaml]: it works for flutter though locally in the terminal with

flutter run --dart-define=ALLOW_DISCORD_RPC=false

even when I remove this kDebug check ofc

I am not sure if dart-define is the same for normal environment variables though
I think we need someone with both bash and flutter skills in here :D

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: You can continue merging this PR. I have borked my installation, but as per my last testing it didn't affect much.

[the remark this answers is not in the document]
Comment (Feichtmeier) [reply snap/snapcraft.yaml]: okay, we can ask some of the canonicals guys on discord if they know how to connect dart-define with the normal environment inside the shell

**Pull request merged**